### PR TITLE
Add enhancements/ directory

### PR DESCRIPTION
**Pull request by dgn**
I added an enhancement template and a first SEP that discusses the process

### CODE REVIEW

File: enhancements/SEP0-template.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+This is free-form; use as many or few sub-headings as appropriate. The following are just examples, use them only if they apply to your SEP.
+
+### User Stories
+
+### API Changes
+
+### Architecture
+
+### Performance Impact
+
```
Comment (sridhargaddam) [inline enhancements/SEP0-template.md:26]: Probably we can also add a section for "Backward Compatibility"

File: enhancements/SEP1-enhancement-process.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+1. conduct our design in the open so that community members can participate and
+1. use asynchronous communication throughout to minimize meeting load.
+
+## Non-goals
+1. Overcomplicating the engineering process. Simple things should stay simple - just submit a PR to fix that bug. It's the hard things we want to make sure to get right using this process. But that doesn't mean you have to write pages of prose. Keep the SEPs short and concise, it will make reviews easier and speed up the process.
+
+## Design
+
+### Applicability
+The SEP process should be used for non-trivial features or epics and any API changes. Changes without user impact will most of the time not need a SEP, but if they're big enough or result in changes to the development workflows (e.g. a rewrite of our integration testing framework) it's probably a good idea to create one.
```
Comment (sridhargaddam) [inline enhancements/SEP1-enhancement-process.md:24]: Work related to an enhancement can sometimes span across multiple repositories:

- Entirely within sail-operator.
- Partially in sail-operator and partially in core Istio (as an example).
- Completely outside of sail-operator.

Should we clarify when a user is expected to use the SEP?

File: enhancements/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+This directory stores all of our Sail Enhancement Proposal documents in markdown format. For more information on the process around enhancements, see [SEP1-enhancement-process.md](./SEP1-enhancement-process.md). If you want to create a SEP, be sure to check out the [SEP template](./SEP0-template.md).
```
Comment (sridhargaddam) [inline enhancements/README.md:1]:
```suggestion
This directory stores all of our Sail Enhancement Proposal (SEP) documents in markdown format. For more information on the process around enhancements, see [SEP1-enhancement-process.md](./SEP1-enhancement-process.md). If you want to create a SEP, be sure to check out the [SEP template](./SEP0-template.md).
```

File: enhancements/SEP1-enhancement-process.md (hunk 2 above)
Comment (dgn) [reply enhancements/SEP1-enhancement-process.md:24]: Good point. I'd argue that the SEP can include potentially needed work in Istio, but they might require yet another design doc if the changeset is big. We would need to link the docs in that case

Comment (dgn) [reply enhancements/SEP1-enhancement-process.md:24]: I added a paragraph

Comment (sridhargaddam) [reply enhancements/SEP1-enhancement-process.md:24]: Thanks @dgn. LGTM, but I'll let others to take a look.

**Review by fjglira — APPROVED** [review]
I like this process, will help to have in one place inside the repo all those big changes that we will do at some point (Instead of having a drive folder with a lot of doc, you can have in the repo)

**Review by fjglira — APPROVED** [review]

**Comment by fjglira** [comment]
I think we need to add also a reference in the read me file to link to this enhancements directory

**Comment by fjglira** [comment]
> LGTM, should we update the [status](https://github.com/istio-ecosystem/sail-operator/blob/main/enhancements/SEP1-enhancement-process.md?plain=1#L3) to Done?

Yes, we can update this at the same time that we add the reference for this documentation in the read me file.

**Pull request merged**